lib/fs/fakefs.go

```diff
@@ -444,13 +504,25 @@ func (fs *fakefs) Rename(oldname, newname string) error {
 		return os.ErrNotExist
 	}
 
-	dst, ok := p1.children[filepath.Base(newname)]
-	if ok && dst.entryType == fakeEntryTypeDir {
-		return errors.New("is a directory")
+	dst, ok := p1.children[newKey]
+	if ok {
+		// in-place case-only directory renames will fail on insensitive systems
+		// they do fail on OS X, so that's expected
+		if dst.entryType == fakeEntryTypeDir {
+			return errors.New("is a directory")
+		}
+
+		if fs.insens && newKey == oldKey {
+			// case-only in-place rename
+			entry.name = filepath.Base(newname)
+			return nil
+		}
```

Comment:
This is sort of the same as the stuff that is happening outside of the if ok condition, we could just make the delete conditional on `fs.insens && p0 == p1 && oldKey == newKey`

Reply:
Still need to circumvent this check for non-destructive in-place rename, so left it here. Tell me if it still looks wrong to you.